flask_testing/utils.py

```diff
@@ -381,15 +378,23 @@ def _spawn_live_server(self):
 
         self._process.start()
 
-        # we must wait for the server to start listening with a maximum timeout of 5 seconds
-        timeout = 5
-        while timeout > 0:
-            time.sleep(1)
+        # We must wait for the server to start listening, with a maximum
+        # timeout of the amount of seconds specified in the LIVESERVER_TIMEOUT
+        # setting. Defaults to 5 seconds.
+        timeout = self.app.config.get('LIVESERVER_TIMEOUT', 5)
+        start_time = time.time()
+        while True:
+            if (time.time() - start_time) > timeout:
+                raise socket.timeout(
+                    'Took more than %s seconds to start the server' % timeout
+                )
+            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
Minor - we can create the socket object once before the loop and keep attempting to connect instead of creating a new one during each iteration